### DISCUSSION TITLE
Casting and Source

### DISCUSSION THREAD

**Discussion by jasonlawless** [upscalebaby/generic-serializable-dictionary#12] [Ideas]
Would it be hard to implement the following?

```csharp
var dic = new Dictionary<string, int>();
var genDic = dic as GenericDictionary<string, int>;
```
and
```csharp
public GenericDictionary<TKey, TValue>(IDictionary<TKey, TValue> source)
```

**Comment by upscalebaby** [comment]
Nice to hear from you again! 

I'm sort of swamped right now so I might not be able to provide something more concrete on this until end of week or perhaps this weekend.

But I suspect it would have the same performance as iterating through the dictionary and adding the entries one by one to the other dictionary.

There might be a way to do it with collection initializers though but I'm not sure.

Btw what is the reason you need a normal dictionary as an intermediary?

**Reply by jasonlawless** [reply]
There is absolutely no rush for this.

The reason this came up was because I had a situation where I needed to verify if all the items in the list were valid against what existed in the scene. And I typically use LINQ for this, and although I could perform the LINQ queries starting from the `GenericDictionary` I could not tie it all back up into it.
I get stuck on `query.ToDictionary(blah blah blah) as GenericDictionary<Ta, Tb>` because it can't cast it back.
Perhaps have a `IEnumerable<TSource>.ToGenericDictionary()` extension for this?

Anyways, I eventually figured out a work around for this, in my case it was easy, but I thought I'd suggest it anyway in case someone (or even me) in the future winds up needing these features as they'd expect from the normal `Dictionary` class.

So just throwing it out there as an optional TODO for ya in case you wanted to make this more flexible/robust ;)

**Reply by upscalebaby** [reply]
Great! Appreciate the suggestion, I will look into it!
